.github/workflows/ci.yml

```diff
@@ -91,14 +133,86 @@ jobs:
+          if [[ "${{ matrix.backend }}" == "dpcpp" ]]; then
+            make_targets="sycl_iterator.pass"
+            ctest_flags="-R sycl_iterator.pass"
+            echo "::warning::dpcpp backend is set. Compile and run only sycl_iterator.pass"
+          fi
           mkdir build && cd build
           cmake -DCMAKE_CXX_STANDARD=${{ matrix.std }} -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
             -DCMAKE_CXX_COMPILER=${{ matrix.cxx_compiler }} -DONEDPL_BACKEND=${{ matrix.backend }} -DONEDPL_DEVICE_TYPE=${{ matrix.device_type }} ..
-          make VERBOSE=1 -j${BUILD_CONCURRENCY}
-          ctest --timeout ${TEST_TIMEOUT} --output-on-failure
+          make VERBOSE=1 -j${BUILD_CONCURRENCY} ${make_targets}
+          ctest --timeout ${TEST_TIMEOUT} --output-on-failure ${ctest_flags}
+
+  windows-testing:
+    name: ${{ matrix.os }}_${{ matrix.cxx_compiler }}_backend=${{ matrix.backend }}_device=${{ matrix.device_type }}_cxx${{ matrix.std }}_${{ matrix.build_type }}
+    runs-on: ['${{ matrix.os }}']
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: windows-2019
+            cxx_compiler: dpcpp
+            std: 17
+            build_type: release
+            backend: tbb
+            device_type: HOST
+          - os: windows-2019
+            cxx_compiler: cl
+            std: 17
+            build_type: release
+            backend: tbb
+            device_type: HOST
+          - os: windows-2019
+            cxx_compiler: dpcpp
+            std: 17
+            build_type: release
+            backend: dpcpp
+            device_type: CPU
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install oneTBB
+        shell: cmd
+        run: |
+          call %CONDA%/condabin/conda.bat activate base
+          conda install -c intel tbb-devel
+      - name: Install oneAPI Compiler
+        if: matrix.cxx_compiler == 'dpcpp' || matrix.cxx_compiler == 'icx'
+        shell: cmd
+        run: |
+          call %CONDA%/condabin/conda.bat activate base
+          conda install -c intel dpcpp_win-64
+      - name: Run testing
+        shell: cmd
+        run: |
+          IF "${{ matrix.backend }}" == "dpcpp" (
+            set ninja_targets=sycl_iterator.pass
+            set ctest_flags=-R sycl_iterator.pass
+            echo ::warning::dpcpp backend is set. Compile and run only sycl_iterator.pass
+          )
+          call %CONDA%/condabin/conda.bat activate base
+          IF "${{ matrix.cxx_compiler }}" == "dpcpp" (
+            powershell $output = dpcpp --version; echo ::warning::$output
+            set toolchain_option=-DCMAKE_TOOLCHAIN_FILE=../cmake/windows-dpcpp-toolchain.cmake
+          )
+          mkdir build && cd build
+          call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
+          set INCLUDE=%CONDA_PREFIX%\Library\include;%INCLUDE%
+          set LIB=%CONDA_PREFIX%\Library\lib;%LIB%
```

Comment:
Are you sure the variables are required to be explicitly set? I'd expect that conda activation sets such kind of variables, doesn't it? What about PATH?

Are these variables needed for all compilers?

Reply:
Yes, these variables are not automatically set by conda (only PATH variable is set) and needed for all compilers, because we need oneTBB in the environment